packages/dev-server/src/MetroDevServer.ts

```diff
@@ -85,7 +86,13 @@ export async function runMetroDevServerAsync(
 
   const metroConfig = await ExpoMetroConfig.loadAsync(projectRoot, { reporter, ...options });
 
-  const { middleware, attachToServer } = createDevServerMiddleware({
+  const {
+    middleware,
+    debuggerProxyEndpoint,
+    messageSocketEndpoint,
+    eventsSocketEndpoint,
+    websocketEndpoints,
+  } = createDevServerMiddleware({
```

Comment:
We may need to check which features are available based on the version installed and implement the new functionality only when the new features are available. Ultimately we'll drop this in SDK 46 when we drop the unversioned CLI.